From 3e1cb12006d535c54ed0dd147121f2f42b2fb615 Mon Sep 17 00:00:00 2001
From: Leon Ma <xindong.ma@intel.com>
Date: Tue, 22 Oct 2013 15:34:31 +0800
Subject: [PATCH 1/2] Add IRQCHIP_SKIP_SET_WAKE flag to ioapic_chip.

BZ: 146212

ioapic_chip structure does not have irq_set_wake inited. -ENXIO will
be returned by set_irq_wake_real() and unbalanced IRQ warnings will
be printed. Adding IRQCHIP_SKIP_SET_WAKE flag so set_irq_wake_real
returns 0 and no warning is printed.

Change-Id: Ie31ec053196c0739b0ee7599f5ce4db1edda5c6d
Signed-off-by: Leon Ma <xindong.ma@intel.com>
(cherry picked from commit 1f7e2255a3bfdfe4e800d6c61887aa58bab1865c)
---
 arch/x86/kernel/apic/io_apic.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/x86/kernel/apic/io_apic.c b/arch/x86/kernel/apic/io_apic.c
index d23aa82..97ead52 100644
--- a/arch/x86/kernel/apic/io_apic.c
+++ b/arch/x86/kernel/apic/io_apic.c
@@ -2522,6 +2522,7 @@ static struct irq_chip ioapic_chip __read_mostly = {
 	.irq_eoi		= ack_apic_level,
 	.irq_set_affinity	= native_ioapic_set_affinity,
 	.irq_retrigger		= ioapic_retrigger_irq,
+	.flags			= IRQCHIP_SKIP_SET_WAKE,
 };
 
 static inline void init_IO_APIC_traps(void)
-- 
1.8.3.2

